### PR TITLE
Final Project complete

### PR DESCRIPTION
**Pull request by gauravPandey88**
1:- Login Functionality completed.
2:- Signup Functionality completed.
3:- Show and Edit profile completed.
4:- Add and remove favourite movie completed.
5:- Show movie detail completed.

**Video**

https://github.com/gauravPandey88/ttnFlix/assets/155935987/46799652-5e8f-4ff0-87e4-382acdf7a049

### **Login Page**
Simulator Screenshot - iPhone14 - 2024-01-26 at 03 42 30

### **Signup page**
Simulator Screenshot - iPhone14 - 2024-01-26 at 03 42 34

**### Home page**
Simulator Screenshot - iPhone14 - 2024-01-26 at 03 38 16

**### Favourite page**
Simulator Screenshot - iPhone14 - 2024-01-26 at 03 38 20

**### Profile Page**
Simulator Screenshot - iPhone14 - 2024-01-26 at 03 38 24

**### Edit profile Page**

Simulator Screenshot - iPhone14 - 2024-01-26 at 03 38 31

### CODE REVIEW

File: lib/EditProfile/screen/edit_profile_screen.dart [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+                                hintStyle: TtnFlixTextStyle
+                                    .defaultTextTheme.titleMedium
+                                    ?.copyWith(color: Colors.grey),
+                                errorText: state.passwordErrorMessage.isNotEmpty ? state.passwordErrorMessage : null,
+                                errorStyle: TextStyle(
+                                    color: Theme.of(context).colorScheme.onError)),
+                          ),
+                        ),
+                    Padding(
+                      padding: const EdgeInsets.only(top: 20.0, left: 10.0),
```
Comment (debabrataroy1) [inline lib/EditProfile/screen/edit_profile_screen.dart]: Manage all padding in a constant file

File: lib/EditProfile/screen/edit_profile_screen.dart [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+                      height: 50,
+                    ),
+                    Center(
+                      child: Container(
+                        height: 48,
+                        width: 250,
+                        decoration: BoxDecoration(
+                            color: TtnflixColors.frozenListYellow
+                                .platformBrightnessColor(context),
+                            borderRadius: BorderRadius.circular(20)),
```
Comment (debabrataroy1) [inline lib/EditProfile/screen/edit_profile_screen.dart]: same for radius

File: lib/EditProfile/screen/edit_profile_screen.dart [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+                                    gender: state.genderType,
+                                    dateofBirth: state.dateofBirth,
+                            password: state.password,
+                            email: state.emailId);
+                            context.router.pop();
+                          },
+                          child: const Text(
+                            'Update',
+                            style: TextStyle(
+                                color: Colors.black, fontSize: 25),
```
Comment (debabrataroy1) [inline lib/EditProfile/screen/edit_profile_screen.dart]: fontSize as well

File: lib/EditProfile/screen/edit_profile_screen.dart [hunk 4]
```diff
@@ -0,0 +1,271 @@
...
+                      ),
+                    ),
+                    Padding(
+                      padding: const EdgeInsets.only(top: 20.0, left: 10.0),
+                      child: Text("Name",
+                          style: TtnFlixTextStyle.defaultTextTheme.titleSmall
+                              ?.copyWith(color: Colors.grey)),
+                    ),
+                    Padding(
+                      padding: const EdgeInsets.only(
```
Comment (AM5324) [inline lib/EditProfile/screen/edit_profile_screen.dart:122]: Add this in constant file

File: lib/EditProfile/screen/edit_profile_screen.dart [hunk 5]
```diff
@@ -0,0 +1,271 @@
...
+                            hintText: "Name",
+                            hintStyle: TtnFlixTextStyle
+                                .defaultTextTheme.titleMedium
+                                ?.copyWith(color: Colors.grey),
+                            errorStyle: TextStyle(
+                                color: Theme.of(context).colorScheme.onError)),
+                      ),
+                    ),
+                    Padding(
+                      padding: const EdgeInsets.only(top: 20.0, left: 10.0),
```
Comment (AM5324) [inline lib/EditProfile/screen/edit_profile_screen.dart]: Add this is constant file

File: lib/EditProfile/screen/edit_profile_screen.dart [hunk 6]
```diff
@@ -0,0 +1,271 @@
...
+                      child: Text("Name",
+                          style: TtnFlixTextStyle.defaultTextTheme.titleSmall
+                              ?.copyWith(color: Colors.grey)),
+                    ),
+                    Padding(
+                      padding: const EdgeInsets.only(
+                          top: 10.0, left: 10.0, right: 10.0),
+                      child: TextField(
```
Comment (AM5324) [inline lib/EditProfile/screen/edit_profile_screen.dart]: textField ki ek reusable class bna lo

File: lib/detail/screen/movie_detail_screen.dart [hunk 7]
```diff
@@ -0,0 +1,271 @@
...
+              padding: const EdgeInsets.all(TtnflixSpacing.spacing8),
+              child: Text(
+                movie.title ?? '',
+                style: TtnFlixTextStyle.defaultTextTheme.headlineLarge
+                    ?.copyWith(
+                        color: TtnflixColors.frozenListYellow
+                            .platformBrightnessColor(context)),
+              ),
+            ),
+            Padding(
```
Comment (AM5324) [inline lib/detail/screen/movie_detail_screen.dart:87]: Icon and text ka ek widget bna lo and RichText bhi use kr sakte ho

File: lib/EditProfile/screen/edit_profile_screen.dart [hunk 8]
```diff
@@ -0,0 +1,271 @@
...
+          builder: (context, state) {
+            if (state is EditProfileLoadedState) {
+              return SingleChildScrollView(
+                  child: Column(
+                      crossAxisAlignment: CrossAxisAlignment.start,
+                      children: <Widget>[
+                    Center(
+                      child: Padding(
+                        padding: const EdgeInsets.all(10.0),
+                        child: Container(
```
Comment (AM5324) [inline lib/EditProfile/screen/edit_profile_screen.dart:65]: height and width put in constant file

**Review by AM5324 — APPROVED** [review]

**Review by debabrataroy1 — APPROVED** [review]

**Pull request merged**